src/main/java/com/team4/testingsystem/services/impl/TestsServiceImpl.java

```diff
@@ -71,6 +68,16 @@ public Test save(Test test) {
 
     @Override
     public long startForUser(long userId, Levels levelName) {
+        List<Test> selfStarted = getByUserId(userId)
+                .stream()
+                .filter((test) -> test.getAssignedAt() == null)
+                .filter((test) -> test.getStartedAt().isAfter(LocalDateTime.now().minusDays(1)))
+                .collect(Collectors.toList());
```

Comment:
1. You can use `.count()` in case you just need a number
2. I think it's better to add an appropriate request from the repository, instead of making the logic in the app.
User can have too many tests and downloading all of them can lead to a huge amount of problems